src/main/java/com/github/checkstyle/regression/module/UnitTestProcessor.java

```diff
@@ -0,0 +1,115 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.regression.module;
+
+import java.io.File;
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.github.checkstyle.regression.data.ModuleInfo;
+import com.puppycrawl.tools.checkstyle.Checker;
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.TreeWalker;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.Configuration;
+import com.puppycrawl.tools.checkstyle.api.FileSetCheck;
+
+/**
+ * Processes the unit test class of a checkstyle module.
+ * This utility class would run {@link UnitTestProcessorCheck} on the file with
+ * the given path and return the collected property info.
+ * @author LuoLiangchen
+ */
+public final class UnitTestProcessor {
+    /** Prevents instantiation. */
+    private UnitTestProcessor() {
+    }
+
+    /**
+     * Processes the unit test class with the given path.
+     * @param path the path of the unit test class
+     * @return the unit test method name to properties map
+     * @throws CheckstyleException          failure when running the check
+     * @throws ReflectiveOperationException failure of reflection on checker and tree walker
+     */
+    public static Map<String, Set<ModuleInfo.Property>> process(String path)
```

Comment:
This class has 2 functions, executing a custom check and getting the results for the specific check.
I assume we may need to write another custom check for https://github.com/checkstyle/regression-tool/issues/6 .

If you agree, I think we should rename this class to something more generic, have it take the module class as input, and return nothing as everything will be retrieved through static getters.

Reply:
Updated.